src/StravaClient.php

```diff
@@ -2,4 +2,12 @@
 
 namespace JordanPartridge\StravaClient;
 
-class StravaClient {}
+use JordanPartridge\StravaClient\Http\Integration\Strava\Strava;
+
+class StravaClient
+{
+    public function __construct(private Strava $strava)
+    {
+        //
+    }
+}
```

Comment:
A lot of the less HTTP and more error handly functinoality can probably be done here.